src/gateways/DraftRegistrationGateway.ts

```diff
@@ -0,0 +1,8 @@
+import { DraftRegistration } from "../entities/DraftRegistration";
+
+export interface DraftRegistrationGateway {
+  deleteUse: (submissionId: string, useIndex: number) => Promise<void>;
+  createEmptyUse: (submissionId: string) => Promise<void>;
+  read: (id) => Promise<DraftRegistration>;
+  update: (id, draftRegistration) => Promise<void>;
```

Comment:
Two new methods to replace the singleton `FormCache` (which was coupled with the concrete `Registration` class, which we're moving away from in favour of plain domain objects).